scipy/stats/_distn_infrastructure.py

```diff
@@ -2879,6 +2901,29 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
                                   locscale_in='loc=0',
                                   # scale=1 for discrete RVs
                                   locscale_out='loc, 1')
+        self._attach_methods()
+        self._construct_docstrings(name, longname, extradoc)
+
+    def __getstate__(self):
+        dct = self.__dict__.copy()
+        # these methods will be remade in __setstate__
+        attrs = ["_parse_args", "_parse_args_stats", "_parse_args_rvs",
+                 "_cdfvec", "_ppfvec", "generic_moment"]
+        [dct.pop(attr, None) for attr in attrs]
+        return dct
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self._attach_methods()
```

Comment:
Actually, can `__get/setstate__` live in `rv_generic` instead of being duplicated in `rv_continuous` and `rv_discrete`?

EDIT. I see, one has `vecentropy` and the other one does not. Why? Anyway, can be left for a follow-up PR.

Reply:
The dynamically created methods `_parse_args`, `_parse_args_stats`, `_parse_args_rvs` need to be removed from all the state dictionaries otherwise there is an exception raised in `__setstate__` when `self.__dict__.update(state)` is called. In addition, rv_continuous has methods like `_cdfvec`  that also need to be removed for the same reason. Thus, each class has it's own `__getstate__`. However, in commit a3ea650cd6240aad3ed321766b7253ac17ecc6a8 I've made `rv_generic.__setstate__` common to all subclasses. This means each class needs to implement an `_attach_methods` method.